x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/ClientTransformIndexer.java

```diff
@@ -196,6 +202,32 @@ protected void doNextBulk(BulkRequest request, ActionListener<BulkResponse> next
         );
     }
 
+    @Override
+    protected void doDeleteByQuery(DeleteByQueryRequest deleteByQueryRequest, ActionListener<BulkByScrollResponse> responseListener) {
+        ClientHelper.executeWithHeadersAsync(
+            transformConfig.getHeaders(),
+            ClientHelper.TRANSFORM_ORIGIN,
+            client,
+            DeleteByQueryAction.INSTANCE,
+            deleteByQueryRequest,
+            responseListener
+        );
+    }
+
+    @Override
+    protected void refreshDestinationIndex(RefreshRequest refreshRequest, ActionListener<RefreshResponse> responseListener) {
```

Comment:
IMO the method is generic enough that it could be called "refreshIndex", without mentioning "destination".